linera-faucet/server/src/lib.rs

```diff
@@ -145,20 +190,29 @@ where
         }
 
         let ownership = ChainOwnership::single(owner);
-        let result = client
+        let result = tmp_client
             .open_chain(ownership, ApplicationPermissions::default(), self.amount)
             .await;
-        self.context.lock().await.update_wallet(&client).await?;
-        let (message_id, certificate) = match result? {
-            ClientOutcome::Committed(result) => result,
-            ClientOutcome::WaitForTimeout(timeout) => {
-                return Err(Error::new(format!(
-                    "This faucet is using a multi-owner chain and is not the leader right now. \
-                    Try again at {}",
-                    timeout.timestamp,
-                )));
+        self.context.lock().await.update_wallet(&tmp_client).await?;
```

Comment:
Why are you updating the wallet even before checking if the chain was opened properly?

Reply:
Good point! That precaution was probably unnecessary in the first place, and now it's even less important because we aren't even using the main chain here.

(Originally my thinking had been: If there is an error that doesn't _necessarily_ mean that no block was created, so better update the wallet, just to be safe. But the faucet is supposed to be the only owner anyway.)